tests/integration/ha_tests/helpers.py

```diff
@@ -378,3 +453,62 @@ async def insert_focal_to_cluster(ops_test: OpsTest) -> None:
     test_collection = db["test_ubuntu_collection"]
     test_collection.insert({"release_name": "Focal Fossa", "version": 20.04, "LTS": True})
     client.close()
+
+
+async def kill_unit_process(ops_test: OpsTest, unit_name: str, kill_code: str):
+    """Kills the DB process on the unit according to the provided kill code."""
+    # killing the only replica can be disastrous
+    app = await app_name(ops_test)
+    if len(ops_test.model.applications[app].units) < 2:
+        await ops_test.model.applications[app].add_unit(count=1)
+        await ops_test.model.wait_for_idle(apps=[app], status="active", timeout=1000)
+
+    # find the DB process on the unit and kill it
+    find_pid = f" run --unit {unit_name} ps aux | grep {DB_PROCESS}"
+    output = await ops_test.juju(*find_pid.split())
+    pid = output[1].split()[1]
+    kill_cmd = f"run --unit {unit_name} -- kill -s {kill_code} {pid}"
```

Comment:
You can simplify this by using `pkill`:

```suggestion
    kill_cmd = f"run --unit {unit_name} -- pkill -s {kill_code} -f {DB_PROCESS}"
```